packages/snjs/__tests__/lib/snCrypto.ts

```diff
@@ -0,0 +1,174 @@
+import {
+  Base64String,
+  HexString,
+  SNPureCrypto,
+  timingSafeEqual,
+  Utf8String,
+} from '@standardnotes/sncrypto-common';
+import { v4 as uuidv4 } from 'uuid';
+
+const CryptoJS = require('crypto-js');
+const sodium = require('libsodium-wrappers');
+
+/**
+ * An SNPureCrypto implementation. Required to create a new SNApplication instance.
+ */
+export default class SNCrypto implements SNPureCrypto {
+  private ready: Promise<void> | null;
+
+  constructor() {
+    this.ready = sodium.ready;
+  }
+
+  public deinit() {
+    this.ready = null;
+  }
+
+  public timingSafeEqual(a: string, b: string) {
+    return timingSafeEqual(a, b);
+  }
+
+  public async pbkdf2(
+    password: Utf8String,
+    salt: Utf8String,
+    iterations: number,
+    length: number
+  ): Promise<HexString | null> {
+    const key = CryptoJS.PBKDF2(password, salt, {
```

Comment:
Just a heads up that for 003 account tests, PBKDF2 will be painfully slow using CryptoJS. Likely will never finish computing with 100,000 iterations.

Reply:
Fixed in https://github.com/standardnotes/snjs/pull/321/commits/97d120f9cf7f99b17e5ec66fc5ce78a87afdefaa